### PR TITLE
calibre: Add python unrardll

### LINKED ISSUE #86152: calibre: No module named unrardll
I'm unable to view cbr files (for example), I get the following Python error:
```
ImportError: No module named unrardll
```

I installed the package with the flag `unrarSupport = true`, I'm using Nixos 20.03 on nixpkgs-unstable. 

@domenkozar @pSub @AndersonTorres 

Maybe I didn't install the package correctly, I'm using a custom overlay where I overrided the `super.calibre`, it seems legit to me.

### PR DESCRIPTION
**Pull request by nyanloutre**
###### Motivation for this change

CBR format was not supported without this lib.
Close  #86152

Also, Calibre will not start without this patch #98197

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by nyanloutre** [comment]
Also it's maybe not useful to change the license type to unfreeRedistributable as the unrar package is now a dependency

**Comment by gebner** [comment]
> Also it's maybe not useful to change the license type to unfreeRedistributable as the unrar package is now a dependency

The license is already an if-then-else.  This is also the reason unrar support is disabled by default.  IIRC there was also a free unrar library that we could use.

File: pkgs/applications/misc/calibre/default.nix [hunk 1]
```diff
@@ -99,7 +99,7 @@ mkDerivation rec {
       sip
       # the following are distributed with calibre, but we use upstream instead
       odfpy
-    ]
+    ] ++ lib.optional (unrarSupport) unrardll
```
Comment (doronbehar) [inline pkgs/applications/misc/calibre/default.nix:102]: This PR won't fix the issue if unrarSupport is `false` by default.

Comment (gebner) [reply pkgs/applications/misc/calibre/default.nix:102]: I am very much against making calibre unfree by default, so unrarSupport should stay false.

Comment (doronbehar) [reply pkgs/applications/misc/calibre/default.nix:102]: Agree. Forgot that rar is unfree.

**Review by veprbl — APPROVED** [review]
Tested calibre build on x86_64-linux (NixOS)

**Pull request merged**